### PR TITLE
flink e2e

### PR DESCRIPTION
**Pull request by MaoZiming**
## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/data/_internal/execution/streaming_executor_state.py [hunk 1]
```diff
@@ -373,16 +373,18 @@ def _replenish_output_budget(self, resource_manager: ResourceManager) -> float:
 
     def lsf_admission_control(self, resource_manager: ResourceManager) -> bool:
```
Comment (frank-lsf) [inline python/ray/data/_internal/execution/streaming_executor_state.py]: let's call this `budget_policy_admission_control` then

File: test_large_e2e_backpressure.py [hunk 2]
```diff
@@ -6,6 +6,7 @@
 import numpy as np
 
 import ray
+import timeline
```
Comment (frank-lsf) [inline test_large_e2e_backpressure.py]: `timeline_utils`

File: variable_duration_banchmark.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
+import datetime
```
Comment (frank-lsf) [inline variable_duration_banchmark.py]: banchmark -> benchmark

**Review by frank-lsf — APPROVED** [review]
Thanks!!

**Comment by MaoZiming** [comment]
Thanks! Fixed

**Pull request merged**